pkg/controllers/deprovisioning/consolidation.go

```diff
@@ -169,23 +222,40 @@ func (c *Consolidation) computeConsolidation(ctx context.Context, node Candidate
 		newNodes[0].Requirements.Add(scheduling.NewRequirement(v1alpha5.LabelCapacityType, v1.NodeSelectorOpIn, v1alpha5.CapacityTypeSpot))
 	}
 
+	validationTimestamp = lo.Ternary(validationTimestamp.IsZero(), c.clock.Now(), validationTimestamp)
 	return Command{
 		nodesToRemove:    []*v1.Node{node.Node},
 		action:           actionReplace,
 		replacementNodes: []*pscheduling.Node{newNodes[0]},
-		createdAt:        c.clock.Now(),
-	}, nil
+	}, validationTimestamp, nil
 }
 
-// validateCommand validates a command for a deprovisioner
-func (c *Consolidation) ValidateCommand(ctx context.Context, nodesToDelete []CandidateNode, cmd Command) (bool, error) {
-	// If we're deleting and there are empty nodes, command is valid.
-	if cmd.action == actionDelete {
-		if c.validateDeleteEmpty(nodesToDelete) {
-			return true, nil
+// waitForTTL is used to wait a consolidationTTL given the initial timestamp of the deprovisioning loop
+func (c *Consolidation) waitForTTL(ctx context.Context, validationTimestamp time.Time) error {
+	remainingDelay := consolidationTTL - c.clock.Since(validationTimestamp)
+	if remainingDelay > 0 {
```

Comment:
Do you need this line or does a `c.clock.After` with a negative value just not wait?

Reply:
Yeah we need it if it's negative. Just tested it and it just gets to the time out.